nmrfx-analyst-gui/src/main/java/org/nmrfx/processor/gui/FXMLControllerManager.java

```diff
@@ -0,0 +1,163 @@
+package org.nmrfx.processor.gui;
+
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import org.nmrfx.analyst.gui.AnalystApp;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * Manages instances of FXMLControllers, from creation to listing them and keeping a way to know which one is the active one.
+ */
+public class FXMLControllerManager {
+    // The active controller is the last one created, or the one that has focus
+    private final SimpleObjectProperty<FXMLController> activeController = new SimpleObjectProperty<>(null);
+
+    // A collection of all known controllers.
+    // It is important to keep insertion order: the first controller is considered the main one and is kept when closing all secondary stages
+    private final Set<FXMLController> controllers = new LinkedHashSet<>();
+
+    /**
+     * Get all registered controllers, keeping the insertion order.
+     *
+     * @return an unmodifiable collection of registered controllers.
+     */
+    public Collection<FXMLController> getControllers() {
+        return Collections.unmodifiableSet(controllers);
+    }
+
+    /**
+     * Register a new controller and mark it as active.
+     *
+     * @param controller the controller to register
+     */
+    private void registerNewController(FXMLController controller) {
+        controllers.add(controller);
+        activeController.set(controller);
+    }
+
+    /**
+     * Unregister a controller. To be called when the associated stage is closed.
+     *
+     * @param controller the controller to unregister.
+     */
+    public void unregister(FXMLController controller) {
+        controllers.remove(controller);
+    }
+
+    /**
+     * Check if a particular controller is already registered.
+     *
+     * @return true if the controller is known.
+     */
+    public boolean isRegistered(@Nullable FXMLController controller) {
+        return controller != null && controllers.contains(controller);
+    }
+
+    /**
+     * Get the active controller, creating one if necessary.
+     *
+     * @return the active controller.
+     */
+    @Nonnull
+    public FXMLController getOrCreateActiveController() {
+        FXMLController active = activeController.get();
+        if (active == null) {
+            //XXX is there actually a case where not having an active controller is a possibility?
+            active = newController();
```

Comment:
I'm not sure if there is, but one thing to check would be to run on mac os. Since I think the File bar can occur separately, I'm not sure if its possible to have all the windows closed but still have the program running with the file menu bar.